adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt

```diff
@@ -1312,6 +1358,76 @@ object AdSdk {
 
     }
 
+    var create: AlertDialog? = null
+    fun showRewardedIntersAd(
+        activity: Activity,
+        adId: String,
+        interstitialCallback: InterstitialCallback
+    ) {
+        var moved = false
+        val ctd = object : CountDownTimer(5000, 1000) {
+            override fun onTick(p0: Long) {
+            }
+
+            override fun onFinish() {
+                if (!moved) {
+                    create?.dismiss()
+                    interstitialCallback.moveNext()
+                    moved = true
+                }
+            }
+        }
+        ctd.start()
+        val builder = AlertDialog.Builder(activity, R.style.DialogTheme)
+        builder.setView(
+            LayoutInflater.from(activity).inflate(R.layout.ad_loading_layout_inters, null)
+        )
+        create = builder.create()
+        create?.show()
+        RewardedInterstitialAd.load(activity, adId,
+            AdRequest.Builder().build(), object : RewardedInterstitialAdLoadCallback() {
+                override fun onAdLoaded(ad: RewardedInterstitialAd) {
+                    ctd.cancel()
+                    if (!moved) {
+                        Log.d(TAG, "Ad was loaded.")
+                        create?.dismiss()
```

Comment:
ad is showing check and is finishing check

Reply:
done